test/test_decimal128.cpp

```diff
@@ -158,24 +164,50 @@ TEST(Decimal128_Query)
         auto rt = db->start_read();
         auto table = rt->get_table("Foo");
         auto col = table->get_column_key("price");
+        auto col_int = table->get_column_key("size");
+        auto col_str = table->get_column_key("description");
         Query q = table->column<Decimal>(col) > Decimal128(0);
         CHECK_EQUAL(q.count(), 99);
         q = table->where().greater(col, Decimal128(0));
         CHECK_EQUAL(q.count(), 99);
         Query q1 = table->column<Decimal>(col) < Decimal128(25);
-        CHECK_EQUAL(q1.count(), 25);
+        CHECK_EQUAL(q1.count(), 24);
         q1 = table->where().less(col, Decimal128(25));
-        CHECK_EQUAL(q1.count(), 25);
+        CHECK_EQUAL(q1.count(), 24);
         q1 = table->where().less_equal(col, Decimal128(25));
-        CHECK_EQUAL(q1.count(), 26);
+        CHECK_EQUAL(q1.count(), 25);
         Query q2 = table->column<Decimal>(col) == realm::null();
         CHECK_EQUAL(q2.count(), 1);
         q2 = table->where().equal(col, realm::null());
         CHECK_EQUAL(q2.count(), 1);
+        q2 = table->where().between(col, Decimal128(25), Decimal128(60));
+        CHECK_EQUAL(q2.count(), 36);
+        Decimal128 sum;
+        Decimal128 max;
+        Decimal128 min(100);
+        size_t cnt = 0;
+        for (auto o : *table) {
+            if (o.get<Int>(col_int) == 3) {
+                auto val = o.get<Decimal128>(col);
+                sum += val;
+                cnt++;
+                if (val > max)
+                    max = val;
+                if (val < min)
+                    min = val;
+            }
+        }
+        CHECK_EQUAL(table->where().equal(col_int, 3).average_decimal128(col), sum / cnt);
+        CHECK_EQUAL(table->where().equal(col_int, 3).maximum_decimal128(col), max);
+        CHECK_EQUAL(table->where().equal(col_int, 3).minimum_decimal128(col), min);
+        CHECK_EQUAL(table->where().equal(col_str, "Nice").average_decimal128(col), Decimal128(57));
+        CHECK_EQUAL(table->where().equal(col_str, "Nice").maximum_decimal128(col), Decimal128(95));
+        CHECK_EQUAL(table->where().equal(col_str, "Nice").minimum_decimal128(col), Decimal128(19));
+        CHECK_EQUAL(table->where().average_decimal128(col), Decimal128(50));
     }
```

Comment:
another nice test would check the min/max/avg of a nullable Decimal128 which only has nulls

Reply:
Done